src/xcode/ENA/ENA/Source/Scenes/Events/Store/MockEventStore.swift

```diff
@@ -0,0 +1,91 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import OpenCombine
+
+class MockEventStore: EventStoring, EventProviding {
+
+	// MARK: - Protocol EventStoring
+
+	func createTraceLocation(_ traceLocation: TraceLocation) -> VoidResult {
+		traceLocationsPublisher.value.append(traceLocation)
+		return .success(())
+	}
+
+	func deleteTraceLocation(id: String) -> EventStoring.VoidResult {
+		traceLocationsPublisher.value.removeAll { $0.guid == id }
+		return .success(())
+	}
+
+	func deleteAllTraceLocations() -> EventStoring.VoidResult {
+		traceLocationsPublisher.value = ([])
+		return .success(())
+	}
+
+	func createCheckin(_ checkin: Checkin) -> EventStoring.IdResult {
+		checkinsPublisher.value.append(checkin)
+		return .success((checkinsPublisher.value.count - 1))
+	}
+
+	func updateCheckin(id: Int, endDate: Date) -> EventStoring.VoidResult {
+		var checkins = checkinsPublisher.value
+		var checkin = checkins[id]
+		checkin.update(checkinEndDate: endDate)
+		checkins.remove(at: id)
+		checkins.insert(checkin, at: id)
+		checkinsPublisher.send(checkins)
+		return .success(())
```

Comment:
I think you can't just use the id as an index here, same for the other `updateCheckIn` method

Reply:
Why not?

Reply:
Yes, didn't thought about deleting. Which would brake the id references. Will change that